src/index.js

```diff
@@ -364,11 +364,13 @@ class Libp2p extends EventEmitter {
    * @returns {Array<Multiaddr>}
    */
   get multiaddrs () {
+    const announceFilter = this._options.addresses.announceFilter || ((multiaddrs) => multiaddrs)
+
     // Filter noAnnounce multiaddrs
     const filterMa = this.addressManager.getNoAnnounceAddrs()
 
     // Create advertising list
-    return this.transportManager.getAddrs()
+    return announceFilter(this.transportManager.getAddrs()
```

Comment:
Looking at this code I realized the announce behavior is not the same as go. Go doesn't concat the listen and announce addresses. If you provide announce addresses, they are the only addresses that get advertised. 

So the behavior should likely be:

- If `announce` addresses are provided, only they will be announced (no filters are applied)
- If `announce` is not provided, the transport addresses will be filtered
  - If `noAnnounce` is provided it will be applied as a filter
  - if the `announceFilter` is provide it will be run

I feel like with `announceFilter` we should just drop support for `noAnnounce`. Downstream projects like js-ipfs can use the filter to do achieve the same end, but with more control.

Additionally, something to think about for js-ipfs that we can use the announceFilter for, is that go-ipfs takes `noAnnounce` from the config and applies those as net masks. It leverages https://github.com/whyrusleeping/multiaddr-filter, so you can more easily filter out ip ranges.

Reply:
yap, I will change this accordingly. I agree on deprecating `noAnnounce`

Reply:
The multiaddr-filter with the net masks is Interesting. I will create an issue in libp2p-utils for us to discuss it better later on.